posts/2021-11-05-emacs-literate-programming-finances.md

```diff
@@ -0,0 +1,273 @@
+
+# Introduction
+
+> Literate programming is a programming paradigm introduced in 1984 by Donald Knuth in which a computer program is given an explanation of its logic in a natural language, such as English, interspersed (embedded) with snippets of macros and traditional source code, from which compilable source code can be generated. The approach is used in scientific computing and in data science routinely for reproducible research and open access purposes. Literate programming tools are used by millions of programmers today.
+> 
+> &#x2014; [1]
+
+This repository holds an example of literate programming used in the finances landscape, implemented with GNU Emacs and some other packages (Org-mode). You can find the complete documentation in reference [6].
+
+The macros used in the program are written in **Python**, a very popular language, and **F#**, which is a functional-first, general purpose, strongly typed, sister language of C# and other .NET implementations.
+
+Also, if you want to really understand and appreciate this document, make sure that you can compile some LaTeX blocks which are presented there. Those blocks were added with the goal of making it easier to understand the problem and the mathematical rationale for the function created later.
+
+If you want to see the compiled version of this document in PDF, check this [link](https://github.com/64J0/literate-prog-finance/blob/master/README.pdf).
+
+
+<a id="org3639a8b"></a>
+
+## Setup
+
+When creating this example I used those software versions:
+
+-   GNU Emacs 28.2.
+-   The *.emacs* configuration mentioned in this repository release: [link](https://github.com/64J0/Emacs-config/releases/tag/v1.0.1).
+
+Disclaimer 1: Some configurations will not work, even if you have the same *.emacs* file, since it will depend on external packages (LaTeX specific packages for example), and the file system structure of your computer.
+
+Disclaimer 2: This document is better understood if you download the *.org* file and open it in Emacs. This way you'll can see the outputs and some org configurations.
+
+
+<a id="orgf7858e1"></a>
+
+# The Problem
+
+In this example, we're going to tackle an investment problem, which could be stated as:
+
+-   Suppose you're going to invest your money into an application that gives you "h%" of this money invested each month, as long as you keep it there (normal investment scenario). During the time you keep it there, you can also add more money (in a month basis), and this money will follow the same rule stated before, but considering that it will start producing more only in the next month that you added it.
+
+To make it more clear, let's use some mathematical notation.
+
+Consider that in the first month, the money you have ($z_0$) is only the initial quantity you decided to invest ($x_0$).
+
+\begin{equation} \tag{1}
+z_0 = x_0
+\end{equation}
```

Comment:
Add an image instead of LaTeX - 1

Reply:
Not 100%, but this issue must solve it in the future:

* https://github.com/64J0/My-blog/issues/44